panoply/datasource.py

```diff
@@ -205,3 +207,24 @@ def wrapper(*args, **kwargs):
 
     return _background_progress
 
+
+def get_nested_data(data, key_path: str) -> List[str]:
+    if not data:
+        return
+    path = key_path if '$.' in key_path else '$.' + key_path
+    expression = parse(path)
+    result = expression.find(data)
+    if len(result) > 1:
+        return [res.value for res in result]
+    if not result:
+        raise KeyError
+    return result[0].value
```

Comment:
```suggestion
    if result:
        return [res.value for res in result]
    raise KeyError
```

Reply:
it  will return None if no data, list of values if there are multiple results found and one value if only one value found
I think we cant just return [res.value for res in result]. 
WDYT?

Reply:
@andrey-panoply but then the data source will need do write ugly code to simply check the returned value. 

How will the usage look like in case the list was returned?